lib/src/lib.rs

```diff
@@ -96,6 +95,10 @@ pub mod rewrite;
 pub mod secret_backend;
 pub mod settings;
 pub mod signing;
+// Allow this file to be outside the normal crate, as it is only used for
+// testing.
+#[path = "../../cli/examples/simple-backend/simple_backend.rs"]
+pub mod simple_backend;
```

Comment:
This looks ugly (and might break `cargo publish`-ed build?)

Maybe we shouldn't move the underlying file until we can remove all `SimpleBackend` dependencies from jj-lib?

Reply:
I agree. Thanks for catching that. This was originally part of the second commit and when the second commit was removed, I didn't realize that parts of it had moved into the first commit. Maybe it was just a mistake.

Reply:
Removed and replaced with a TODO which mentions that we should do that.